selfhost/bound.wyv

```diff
@@ -43,6 +43,7 @@ datatype Exp
     Field(receiver:Exp, field:String)
     Val(binding:Binding, exp:Exp)
     TypeDecl(name:Binding, members:List[types.DeclType])
+    NewType(binding:Binding, body:List[Exp])
```

Comment:
Let's just call this New, because this is an Expression, not a Type.  (could also be NewExpr, but we're not using the Expr suffix elsewhere).